dataset/batch_image.py

```diff
@@ -1,525 +1,717 @@
+            - 'random' - place the upper-left corner of the input image on the randomly sampled position
+                         in the background. Position is sampled uniformly such that there is no need for cropping.
+            - other - place the upper-left corner of the input image on the given position in the background.
+        background_shape : sequence
+            shape of the background image.
+
+        Returns
+        -------
+        sequence : calculated origin in the form (row, column)
+        """
+        if isinstance(origin, str):
+            if origin == 'top_left':
+                origin = 0, 0
+            elif origin == 'center':
+                origin = np.maximum(0, np.asarray(background_shape) - image_shape) // 2
+            elif origin == 'random':
+                origin = (np.random.randint(background_shape[0]-image_shape[0]+1),
+                          np.random.randint(background_shape[1]-image_shape[1]+1))
+        return np.asarray(origin, dtype=np.int)
+
+    def _scale_(self, image, factor, preserve_shape=False, origin='center'):
+        """ Scale the content of each image in the batch.
+
+        Resulting shape is obtained as original_shape * factor.
+
+        Parameters
+        -----------
+        components : str
+            component to get an image from
+        p : float
+            probability of applying the transforms
+        image : np.ndarray
+            image to scale
+        factor : float, sequence
+            resulting shape is obtained as original_shape * factor
+            - float - scale all axes with the given factor
+            - sequence (factor_1, factort_2, ...) - scale each axis with the given factor separately
+
+        preserve_shape : bool
+            whether to preserve the shape of the image after scaling
+
+        origin : {'center', 'top_left', 'random'}, sequence
+            Relevant only if `preserve_shape` is True.
+            Position of the scaled image with respect to the original one's shape.
+            - 'center' - place the center of the rescaled image on the center of the original one and crop
+                         the rescaled image accordingly
+            - 'top_left' - place the upper-left corner of the rescaled image on the upper-left of the original one
+                           and crop the rescaled image accordingly
+            - 'random' - place the upper-left corner of the rescaled image on the randomly sampled position
+                         in the original one. Position is sampled uniformly such that there is no need for cropping.
+            - sequence - place the upper-left corner of the rescaled image on the given position in the original one.
+
+        Returns
+        -------
+        np.ndarray : rescaled image
+        """
+        if np.any(np.asarray(factor) <= 0):
+            raise ValueError("factor must be greater than 0")
+        image_shape = image.shape[:-1]
```

Comment:
Create a function to get an image shape and use it everywhere

Reply:
+